web/src/lib/actions/single-grid-row.ts

```diff
@@ -0,0 +1,40 @@
+export type Options = {
+  onChange: (itemCount: number) => void;
+};
+
+function getGridGap(element: Element) {
+  const style = getComputedStyle(element);
+
+  return {
+    columnGap: parsePixels(style.columnGap),
+  };
+}
+
+const parsePixels = (style: string) => Number.parseInt(style, 10) || 0;
+
+export function singleGridRow(container: HTMLElement, { onChange }: Options) {
```

Comment:
Why is this better than just using a wrapper component instead? In general I don't really like the `use:action`. It seems harder to debug, troubleshoot, and understand, but that may just be my lack of understanding. I can easily understand `<SingleGridRow let:container>` though. Is it just a matter of data accessibility? Accessing and controlling the width/visibility of dynamic children?

Reply:
I generally don't mind actions, but after rewriting it as a component it seems the better option. You loose the flexibility of applying actions to arbitrary elements, but that's not really an issue in this case.